include/pybind11/pybind11.h

```diff
@@ -1378,14 +1378,32 @@ class generic_type : public object {
                                   handle fset,
                                   detail::function_record *rec_func) {
         const auto is_static = (rec_func != nullptr) && !(rec_func->is_method && rec_func->scope);
-        const auto has_doc = (rec_func != nullptr) && (rec_func->doc != nullptr)
-                             && pybind11::options::show_user_defined_docstrings();
+
+        std::string doc;
+        if (rec_func != nullptr) {
+            if (pybind11::options::show_function_signatures()) {
+                doc += name;
+                if (rec_func->signature != nullptr) {
+                    std::string sig = rec_func->signature;
+                    size_t ret = sig.rfind(" -> ");
```

Comment:
One could extend function_record with an additional member "return_type" or similar, but it would only be used for properties.
This appeared to invasive to me, so I did it the same as for constructors:
https://github.com/pybind/pybind11/blob/master/include/pybind11/pybind11.h#L1046